### PR TITLE
Filters: Reset page to 0 on filter change

### PR DESCRIPTION
**Pull request by rickisXP**
Closes https://github.com/iFixit/react-commerce/issues/1005, https://github.com/iFixit/react-commerce/issues/697

Issue (description from #1005)
--
When on a device page (like https://www.ifixit.com/Parts/iPhone_6s?p=2), and you click an item type (like Case Components), the item type is applied but the other facets remain, and consequently sometimes there are no results.

QA
--

1. Go to a device page with multiple pages
2. Navigate to the second page
3. Apply an `Item Type` filter
4. Ensure that the newly filtered page is no longer on the second page

cr_req 2

### CODE REVIEW

**Review by masonmcelvain — DISMISSED** [review]
CR ✌🏻 Woohoo!

File: frontend/templates/product-list/sections/FilterableProductsSection/RefinementSingleSelect.tsx [hunk 1]
```diff
@@ -168,8 +168,7 @@ const SingleSelectItem = React.memo(function SingleSelectItem({
                   delete refinementList[attribute];
                }
                return {
-                  ...prevUiState,
-                  refinementList: refinementList,
+                  refinementList,
```
Comment (danielbeardsley) [inline frontend/templates/product-list/sections/FilterableProductsSection/RefinementSingleSelect.tsx:171]:
```suggestion
                  ...prevUiState,
                  refinementList,
                  page: 0,
```

There may be a number of other things in the UI state. I'm not sure we want to get rid of them all if we're only trying to set the page to 0. Thoughts?

Comment (masonmcelvain) [reply frontend/templates/product-list/sections/FilterableProductsSection/RefinementSingleSelect.tsx:171]: Yea I could see an argument for keeping the text in the search box across refinements (which is the only other thing we use the UiState for currently). Does that sound reasonable? @sterlinghirsh

Comment (sterlinghirsh) [reply frontend/templates/product-list/sections/FilterableProductsSection/RefinementSingleSelect.tsx:171]: The search should definitely remain.

**Review by danielbeardsley — APPROVED** [review]
CR :+1:

File: frontend/templates/product-list/sections/FilterableProductsSection/RefinementSingleSelect.tsx (hunk 1 above)
Comment (danielbeardsley) [reply frontend/templates/product-list/sections/FilterableProductsSection/RefinementSingleSelect.tsx:172]: Ok, committed this suggestion then.

**Comment by sterlinghirsh** [comment]
CR :zap:

**Review by masonmcelvain — DISMISSED** [review]
CR ✌🏻 

Though when on /Parts/iPhone/Adhesives?p=2, and you click the `x` on the blue "Adhesives" flair box, the `p=` filter remains (like @deltuh-vee described earlier).

[Screencast from 11-10-2022 08:15:41 AM.webm](https://user-images.githubusercontent.com/52104630/201149304-d9f08f42-7627-4ef6-b982-b1bca7e3e7e4.webm)

(sorry the video is a little choppy, the UI was smoother in practice)
https://react-commerce-eubttwjna-ifixit.vercel.app/Parts/iPhone/Adhesives?p=2

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 

> Setting it to 0 results in a url like "?p=" which is kinda ugly.

This behavior wasn't happening for me, but this seems to work too 👍🏻

**Comment by deltuh-vee** [comment]
Is Item Type the only filter we want to reset the page results?

Steps to reproduce:
1. Visit https://react-commerce-git-reset-page-with-filter-ifixit.vercel.app/Parts
2. Expand Device Type
3. Select iPhone
4. Navigate to the fifth page of results
5. Remove the filter
6. Notice the page didn't revert back to 0

**Comment by masonmcelvain** [comment]
> Is Item Type the only filter we want to reset the page results?

No, all single select facets should reset the `p=` when the facet is selected, and when the facet is deselected.

However, I'm not able to reproduce your results.

https://user-images.githubusercontent.com/52104630/201190693-97060e9f-48f6-4ed4-a595-d82a72cea1a2.mp4

(Sorry the screen recording is a little sloppy. Ubuntu's native screen recording isn't working for me ATM 🤪 )

**Comment by deltuh-vee** [comment]
I used the pill to deselect the filter.
https://user-images.githubusercontent.com/5289715/201191586-4203bab9-cdc3-48b4-9725-5cfaaf50723e.mp4

**Comment by masonmcelvain** [comment]
Ah, gotcha. Yes, I think the pill is still broken for all single selects, I don't think there's been a code change for that yet

dev_block ✌🏻 un_deploy_block ✌🏻

**Comment by sterlinghirsh** [comment]
multiselects should also reset the page.

**Comment by danielbeardsley** [comment]
This seems like it was mostly fixed by the recent #1004 because we swtiched to using `useMenu()` which auto-resets the page number on filter changes. I'm gonna do some QAing.

**Comment by rickisXP** [comment]
I went ahead and merged in changes from master, and set the page to `NaN` when a filter is removed using the pill. un_dev_block :+1: 

deploy_block :thought_balloon: on confirming whether these console warnings are okay to ignore or not?
image

**Comment by sterlinghirsh** [comment]
I'd rather not ignore them. @danielbeardsley what was the result of your QAing?

**Comment by dhmacs** [comment]
We had that warning since we started using Algolia hooks basically. I've spent some time trying to debug it, but it's still not clear how to fix it. I've haven't looked into it further since it doesn't seem to cause bugs, so it was not top priority on my list. 
Perhaps it's something that will be fixed by a new version of the library.

**Comment by masonmcelvain** [comment]
@dhmacs is right, that warning is has been present on `main` for some time. un_deploy_block ✌🏻

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻

**Comment by danielbeardsley** [comment]
The page number is reset now in all the cases we care about, but other filters (multi-select) are not reset when choosing another item-type. If that is important we should make a new issue.

**Comment by masonmcelvain** [comment]
Ah right, forgot about multi-select. @sterlinghirsh said it was important here: https://github.com/iFixit/react-commerce/pull/1010#issuecomment-1310918022

I made an issue: https://github.com/iFixit/react-commerce/issues/1034

**Pull request closed without merging**